app/controllers/students/notes_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Students::NotesController < NotesController
+  before_filter :set_notable
+  
+  def new
+    @note = @notable.notes.new
+  end
+
+  private
+
+  def set_notable
+    @notable = Student.find(params[:student_id])
+  end
```

Comment:
it's usually hard to tell if an abstraction is right if you only have one instance of it. Usually it's a good idea to wait to do the abstraction until you have multiple, that way it's easy to confirm if `NotesController` is correctly abstracted.

So I suggest you pull everything into this controller and delete the `NotesController` and then only do the abstraction in Task 18, maybe even only Task 19. I think this will also solve the controller spec issue.